sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,851 @@
+  DigitalTwinsAddRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipResponse,
+  DigitalTwinsDeleteRelationshipOptionalParams,
+  DigitalTwinsListIncomingRelationshipsResponse,
+  DigitalTwinsListRelationshipsOptionalParams,
+  DigitalTwinsListRelationshipsResponse,
+  DigitalTwinsSendTelemetryOptionalParams,
+  DigitalTwinsSendComponentTelemetryOptionalParams,
+  DigitalTwinModelsGetByIdResponse,
+  DigitalTwinModelsListOptionalParams,
+  DigitalTwinModelsListResponse,
+  DigitalTwinModelsListOptions,
+  DigitalTwinModelsAddResponse,
+  DigitalTwinModelsAddOptionalParams,
+  ModelData,
+  IncomingRelationship,
+  EventRoutesGetByIdResponse,
+  EventRoutesListOptionalParams,
+  EventRoutesListNextResponse,
+  EventRoute,
+  EventRoutesAddOptionalParams,
+  QueryQueryTwinsResponse,
+  QuerySpecification
+} from "./generated/src/models";
+
+/**
+ * Client for Azure IoT DigitalTwins API.
+ */
+export class DigitalTwinsClient {
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated AzureDigitalTwinsAPI
+   */
+  private readonly client: GeneratedClient;
+
+  // private readonly client!: AzureDigitalTwinsAPI;
+
+  /**
+   * Creates an instance of AzureDigitalTwinsAPI.
+   *
+   * Example usage:
+   * ```ts
+   * const { DigitalTwinsClient, ServiceClientCredentials } = require("@azure/digitaltwins");
+   *
+   * const client = new DigitalTwinsClient(
+   *   "<endpoint>",
+   *   new DefaultAzureCredential();
+   * );
+   * ```
+   * @param {string} endpoint The endpoint of the service. If options parameter is given than this value will be ignored.
+   * @param {TokenCredential | ServiceClientCredentials} credential Used to authenticate requests to the service.
+   * @param {AzureDigitalTwinsAPIOptions} [options] Used to configure the service client.
+   */
+  constructor(
+    enpoint: string,
+    credential: TokenCredential | ServiceClientCredentials,
+    options?: AzureDigitalTwinsAPIOptions
+  ) {
+    if (!options) {
+      const options: AzureDigitalTwinsAPIOptions = {};
+      options.baseUri = enpoint;
+    }
+
+    this.client = new GeneratedClient(credential, options);
+  }
+
+  /**
+   * Get a digital twin
+   *
+   * @param digitalTwinId The Id of the digital twin.
+   * @returns The application/json digital twin and the http response.
+   */
+  public getDigitalTwin(digitalTwinId: string): Promise<DigitalTwinsGetByIdResponse> {
+    return this.client.digitalTwins.getById(digitalTwinId);
+  }
+
+  /**
+   * Create or update a digital twin
+   *
+   * @param digitalTwinId The Id of the digital twin to create or update.
+   * @param digitalTwin The application/json digital twin to create.
+   * @param enableUpdate If true then update of an existing digital twin is enabled.
+   * @returns The created application/json digital twin and the http response.
+   */
+  public upsertDigitalTwin(
+    digitalTwinId: string,
+    digitalTwin: string,
```

Comment:
if this is a JSON blob, maybe the name should be `digitalTwinJson`? 

Also, would it be useful to take the JSON arguments as objects that the client can serialize?

Reply:
I changed the name as suggested. I think the type should stay string as that is the most convenient type in Node on top of "any" type in the generated code. I mean if the service Json model will change we don't need to maintain that.

---
In reply to: [447260881](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r447260881) [](ancestors = 447260881)